gqlalchemy/memgraph.py

```diff
@@ -144,3 +166,147 @@ def new_connection(self) -> Connection:
+            self.load_node_with_id(node)
+        else:
+            return self.load_node_with_all_properties(node)
+
+    def load_node_with_all_properties(self, node: Node) -> Optional[Node]:
+        results = self.execute_and_fetch(
+            f"MATCH (node: {node._label} WHERE {node._get_cypher_fields_or_block('node')} RETURN node"
+        )
+        return self.get_variable_assume_one(results, "node")
+
+    def load_node_with_id(self, node: Node) -> Optional[Node]:
+        results = self.execute_and_fetch(f"MATCH (node: {node._label}) WHERE id(node) = {node._id} RETURN node")
+
+        return self.get_variable_assume_one(results, "node")
+
+    def load_relationship(self, relationship: Relationship) -> Optional[Relationship]:
+        if relationship._id is not None:
+            return self.load_relationship_with_id(relationship)
+        elif relationship._start_node_id is not None and relationship._end_node_id is not None:
+            return self.load_relationship_with_start_node_id_and_end_node_id(relationship)
+        else:
+            return self.load_relationship_with_all_properties(Relationship)
+
+    def load_relationship_with_id(self, relationship: Relationship) -> Optional[Relationship]:
+        results = self.execute_and_fetch(
+            f"MATCH ()-[relationship: {relationship._type}]->() WHERE id(relationship) = {relationship._id} RETURN relationship"
+        )
+        return self.get_variable_assume_one(results, "relationship")
+
+    def load_relationship_with_start_node_id_and_end_node_id(
+        self, relationship: Relationship
+    ) -> Optional[Relationship]:
+        results = self.execute_and_fetch(
+            f"MATCH (start_node)-[relationship:{relationship._type}]->(end_node) WHERE id(start_node) = {relationship._start_node_id} AND id(end_node) = {relationship._end_node_id} RETURN relationship"
+        )
+        return self.get_variable_assume_one(results, "relationship")
+
+    def load_relationship_with_start_node_and_end_node(
+        self, relationship: Relationship, start_node: Node, end_node: Node
+    ) -> Optional[Relationship]:
+        results = self.execute_and_fetch(
+            f"MATCH (start_node: {start_node._label})-[relationship:{relationship._type}]->(end_node: {end_node._label}) WHERE id(start_node) = {start_node._id} AND id(end_node) = {end_node._id} RETURN relationship"
+        )
+        return self.get_variable_assume_one(results, "relationship")
+
+    def load_relationship_with_all_properties(self, relationship: Relationship) -> Optional[Relationship]:
+        where_block = relationship._get_cypher_fields_or_block("relationship")
+        results = self.execute_and_fetch(
+            f"MATCH ()-[relationship: {relationship._type}]->() WHERE {where_block} RETURN relationship"
+        )
+        return self.get_variable_assume_one(results, "relationship")
+
+    def save_relationship(self, relationship: Relationship) -> Optional[Relationship]:
```

Comment:
This can be avoided by not being able to update relationshis without internal id. Before updating a relationship you first have to create it or load it from memgraph.